### DISCUSSION TITLE
Voice FX

### DISCUSSION THREAD

**Discussion by dukenewcombe** [EDCD/EDDI#2306] [EDDI Standalone Q&A] [answered]
We have probably all noted that there are a number of "radio" calls that play in the background as you approach a station or sit in the hanger and so on. 

Currently EDDI have awesome adjustments for TTS, albeit a tad limited, so I thought I'd suggest expanding on that a little. In my situation I use VA for all my macro's and triggers, although I don't like using voice triggers as VA seems to randomly recognise words when nothing was said... making it unreliable for that... but it's great at macros.  To add to the immersion I include "radio" like calls to and from the stations, which I currently apply the distortion effect within VA to for another layer of realism... but that filter just doesn't cut the mustard for me. 

Indeed when I am in the SRV or on foot, I'd rather have my AI talking over comms and not be in my helmet as it were.

Is this something that's easily achievable?

**Comment by Tkael** [comment, marked as answer]
Are you familiar with EDDI's Transmit() function?
https://github.com/EDCD/EDDI/wiki/Help#transmit

It can also be invoked from the plugin interface in VoiceAttack.
https://github.com/EDCD/EDDI/wiki/VoiceAttack-Integration#transmit

**Reply by dukenewcombe** [reply]
Thanks mate,

Yes I have, but admittedly I have failed to use that... I just tested it, certainly an improvement on my VA version as it has that tinny effect it just lacks the background static effect. Again that is just my opinion... 

could there be a couple of additional sliders for "transmit" for tinny and static effects perhaps? I'm very new to C# but I might have a tinker with a repo.

**Reply by Tkael** [reply]
Hmm... I'm not sure how to add a static popping effect but we could investigate that.

That said, I would expect that popping effect to persist for as long as the radio connection were open (so potentially the entire time that I was not in my ship rather than just when speech was occurring)?

If the sound should persist as long as you are in an SRV or helmeted suit then you may want to implement that as a separate sound layer rather than as a part of a speech effect.
